WooCommerce/Classes/ViewRelated/Hub Menu/HubMenuViewController.swift

```diff
@@ -31,6 +31,25 @@ final class HubMenuViewController: UIHostingController<HubMenu> {
     func showPaymentsMenu() {
         show(InPersonPaymentsMenuViewController(), sender: self)
     }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+
+        // We want to hide navigation bar *only* on HubMenu screen. But on iOS 16, the `navigationBarHidden(true)`
+        // modifier on `HubMenu` view hides the navigation bar for the whole navigation stack.
+        // Here we manually hide or show navigation bar when entering or leaving the HubMenu screen.
+        if #available(iOS 16.0, *) {
+            self.navigationController?.setNavigationBarHidden(true, animated: animated)
+        }
+    }
+
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+
+        if #available(iOS 16.0, *) {
+            self.navigationController?.setNavigationBarHidden(false, animated: animated)
+        }
```

Comment:
nit: does it make sense to reverse the order of the `super` call since this is a "disappear" event?

```suggestion       
        if #available(iOS 16.0, *) {
            self.navigationController?.setNavigationBarHidden(false, animated: animated)
        }
        
        super.viewWillDisappear(animated)
```

Reply:
[The API doc](https://developer.apple.com/documentation/uikit/uiviewcontroller/1621485-viewwilldisappear) says
> If you override this method, you must call super at some point in your implementation. 

I don't think the order of these two statements would make any difference, the pop animation should begin and end at the same time.